### PR TITLE
Fix a flaky test - test_that_forbidden_403_error_does_not_trigger_log…

### PR DESCRIPTION
**Pull request by potrebic**
…out_notification_for_non_token_requests

This test was flaky due to a subtle (and still not well understood)
interaction between OCMock's 'reject' method and the BOXAPIOperation
finish method. This resulted in random crashes in the finish method.

The solution was to use a BOXAPIOperation `spy` implementation allowing
a test to block until the finish method is *completely* finished. This
technique eliminates the random crashing.

Comments in the code have more details.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @potrebic has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDKTests/BOXRequestTests.m [hunk 1]
```diff
@@ -222,8 +262,6 @@ - (void)test_that_forbidden_403_error_does_not_trigger_logout_notification_for_n
         [expectation fulfill];
     }];
     [self waitForExpectationsWithTimeout:2.0 handler:nil];
-
-    [operationMock verify];
```
Comment (potrebic) [inline BoxContentSDK/BoxContentSDKTests/BOXRequestTests.m:226]: NOTE - this was removed because it was unnecessary. The `reject` call above sets up a fail-fast scenario, when the test is failed immediately if `sendLogoutNotification` is called. And there are no other OCMock expectations... thus calling verify is a no-op. So I removed. All this is unrelated to fixing the flakiness of this test.

**Review by jlawton — APPROVED** [review]

**Review by MashfiqueA — APPROVED** [review]
